### PR TITLE
[TT][LB] fixing the test-only endpoint to clear the DB

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

File: app/uk/gov/hmrc/ctc/config/AppConfig.scala [hunk 1]
```diff
@@ -38,11 +38,6 @@ class AppConfig @Inject()(val runModeConfiguration: Configuration, environment:
     runModeConfiguration.getStringList(key).map(l => l.toList).getOrElse(throw new Exception(s"Missing configuration key: $key"))
   }
 
-  lazy val withBackend: Boolean = {
```
Comment (robertbutacu) [inline app/uk/gov/hmrc/ctc/config/AppConfig.scala]: I still believe we should have this functionality unless development is rather stable - its a quick fix to release any dependencies between BE/FE.

Maybe discuss with Giuseppe as well and see what are his opinions - I don't mind either way.

File: app/uk/gov/hmrc/ctc/controllers/test/TestController.scala [hunk 2]
```diff
@@ -27,13 +27,13 @@ import scala.concurrent.ExecutionContext
 class TestController @Inject()(testService: TestService)(implicit ec: ExecutionContext) extends BaseController {
 
   def clearDB: Action[AnyContent] = Action.async { implicit request =>
-    testService.clearDB.map(_ => NoContent).recover {
```
Comment (robertbutacu) [inline app/uk/gov/hmrc/ctc/controllers/test/TestController.scala]: This is not related to this per se, but with the way the connectors have been split and my lack of understanding as to why this happened - I would understand the separation on a service-level, but why did it happen on the connector level instead ( report/transit connectors )?

File: app/uk/gov/hmrc/ctc/config/AppConfig.scala (hunk 1 above)
Comment (udinobi) [reply app/uk/gov/hmrc/ctc/config/AppConfig.scala]: At this point we even have (test) endpoints for feeding the DB. That should more than enough now, as well as to always running with the real back-end, in order to not inadvertently break anything.

Comment (peppe-hmrc) [reply app/uk/gov/hmrc/ctc/config/AppConfig.scala]: from my side is better removing it and having an actual failure

File: app/uk/gov/hmrc/ctc/controllers/test/TestController.scala (hunk 2 above)
Comment (udinobi) [reply app/uk/gov/hmrc/ctc/controllers/test/TestController.scala]: Just to have a clean logic separation.

Comment (peppe-hmrc) [reply app/uk/gov/hmrc/ctc/controllers/test/TestController.scala]: I'm not sure I understood correctly the separation but I think is about separating the report responsibility form the transit one. I'm expecting that we will have more report page to provide in the future to the managers. 
It like thinking too ahead but at the same time looks an easy change done now

Comment (robertbutacu) [reply app/uk/gov/hmrc/ctc/controllers/test/TestController.scala]: Yes, but I expected this to happen on the service layer - at the connector level, its just a call - I see this layer where the context is based on the service being called, not what logic is being applied on the data, while at the service layer there is a pretty strong case of separating the concerns already.

Comment (peppe-hmrc) [reply app/uk/gov/hmrc/ctc/controllers/test/TestController.scala]: I see the point here ... lets see what will happen, if the complexity of the two connector is very basic we might think to merge together

**Pull request merged**